lib/paranoia.rb

```diff
@@ -102,12 +102,12 @@ def paranoia_delete
 
   def restore!(opts = {})
     self.class.transaction do
-      run_callbacks(:restore) do
-        recovery_window_range = get_recovery_window_range(opts)
-        # Fixes a bug where the build would error because attributes were frozen.
-        # This only happened on Rails versions earlier than 4.1.
-        noop_if_frozen = ActiveRecord.version < Gem::Version.new("4.1")
-        if within_recovery_window?(recovery_window_range) && ((noop_if_frozen && !@attributes.frozen?) || !noop_if_frozen)
+      recovery_window_range = get_recovery_window_range(opts)
+      # Fixes a bug where the build would error because attributes were frozen.
+      # This only happened on Rails versions earlier than 4.1.
+      noop_if_frozen = ActiveRecord.version < Gem::Version.new("4.1")
```

Comment:
Seems we could remove this? This will never be true anymore.

Reply:
I don't think we need it, but it's not hurting us. I'd rather make rebasing with upstream easier if it's not going to change anything for us